classes/01class/exercises/c01-aws02/fabioacp/README.md

```diff
@@ -0,0 +1,196 @@
+# C01-AWS02
+
+## Commands Execution Output
+
+- Commands to create the S3 Bucket and to copy the file to the bucket:
+```bash
+# Create the bucket
+❯ aws s3 mb s3://da-c01-aws02 --region ap-southeast-2
+make_bucket: da-c01-aws02
+
+# The following command lists all buckets
+❯ aws s3 ls                  
+2020-06-16 22:40:17 da-c01-aws02
+
+# Create the file.txt
+❯ touch file.txt
+
+# Upload the file.txt to s3 bucket
+❯ aws s3 cp file.txt s3://da-c01-aws02/
+upload: ./file.txt to s3://da-c01-aws02/file.txt
+
+# The following command lists all objects and folders
+❯ aws s3 ls s3://da-c01-aws02
+2020-06-16 22:41:54          0 file.txt
+
+```
+
+- Commands to allow the EC2 instance to access the files in S3:
+```bash
+
+# 1 - Create the following trust policy and save it in a text file named ec2-role-trust-policy.json.
+
+❯ cat <<EOF > ec2-role-trust-policy.json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": { "Service": "ec2.amazonaws.com"},
+      "Action": "sts:AssumeRole"
+    }
+  ]
+} 
+EOF
+
+❯ ls
+ec2-role-trust-policy.json
+
+# 2 - Create the s3access role and specify the trust policy that you created using the create-role command.
+
+❯ aws iam create-role --role-name s3access --assume-role-policy-document file://ec2-role-trust-policy.json
+{
+    "Role": {
+        "Path": "/",
+        "RoleName": "s3access",
+        "RoleId": "AROA6HNPHMJJEKKWM6AHY",
+        "Arn": "arn:aws:iam::978009416274:role/s3access",
+        "CreateDate": "2020-06-18T12:21:47Z",
+        "AssumeRolePolicyDocument": {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "Service": "ec2.amazonaws.com"
+                    },
+                    "Action": "sts:AssumeRole"
+                }
+            ]
+        }
+    }
+}
+
+# 3 - Create an access policy and save it in a text file named ec2-role-access-policy.json. 
+
+❯ cat <<EOF > ec2-role-access-policy.json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:Get*",
+                "s3:List*"
+            ],
+            "Resource": "*"
```

Comment:
Can you use some restricted permissions to make this config more secure?

Reply:
done